piecrust/src/imports.rs

```diff
@@ -46,6 +48,40 @@ impl DefaultImports {
     }
 }
 
+fn check_ptr(
+    instance: &WrappedInstance,
+    offset: u32,
+    len: u32,
+) -> Result<(), Error> {
+    let mem_len = instance.with_memory(|mem| mem.len());
+
+    let offset = offset as usize;
+    let len = len as usize;
+
+    if offset + len > mem_len {
```

Comment:
Since `offset` and `len` are both originally `u32`, `usize` will be - in a 64-bit platform - never be overflown even if both are `u32::MAX`.

We already don't really support 32bit platforms due to hoe memory management happens.